src/main/java/apoc/mongodb/MongoDBColl.java

```diff
@@ -67,31 +74,76 @@ public void close() throws IOException {
      * @return
      */
     private Map<String, Object> documentToPackableMap(Map<String, Object> document) {
+        return (Map<String, Object>) convertAndExtract(document);
+    }
+
+    public Object convertAndExtract(Object data) {
+        if (data == null) {
+            return null;
+        }
+        if (data instanceof Map) {
+            Map<String, Object> map = (Map<String, Object>) data;
+            return map.entrySet().stream()
+                    .map(e -> {
+                        Object value;
+                        if (ID.equals(e.getKey())) { // avoid circular conversions
+                            if (compatibleValues && objectIdAsMap) {
+                                try {
+                                    value = jsonMapper.readValue(jsonMapper.writeValueAsBytes(e.getValue()), Map.class);
```

Comment:
will we lose the object-id here? so that e.g. extractReference won't work anymore?
Or will it be a map of _id to object-id ? would probably good to add some more examples to the docs perhaps in a separate PR

Reply:
That behavoiur is related on how we convert the `_id` field:

Follwing an example that could help to understand the behaviour:

Given the following collections:

```
// Product
...
{"_id": ObjectId("product1"), "name": "Product 1", "price": 100}
{"_id": ObjectId("product3"), "name": "Product 2", "price": 200}
{"_id": ObjectId("product3"), "name": "Product 3", "price": 300}
...
```

```
// Person
...
{"_id": ObjectId("person"), "name": "Andrea", "bought": [ObjectId("product1"), ObjectId("product3")]}
...
```

With the `extractReferences=true`, `compatibleValues=true` and `objectIdAsMap=false`:

```
{
  "_id": "person",
  "name": "Andrea",
  "bought": [
    {"_id": "product1", "name": "Product 1", "price": 100},
    {"_id": "product3", "name": "Product 3", "price": 300}
  ]
}
```


With the `extractReferences=true`, `compatibleValues=true` and `objectIdAsMap=true`:

```
{
  "_id": {
  	"timestamp": <...>,
	"machineIdentifier": <...>,
	"processIdentifier": <...>,
	"counter": <...>,
  },
  "name": "Andrea",
  "bought": [
    {
      "_id": {
	  	"timestamp": <...>,
		"machineIdentifier": <...>,
		"processIdentifier": <...>,
		"counter": <...>,
	  },
	  "name": "Product 1",
	  "price": 100
	},
    {
      "_id": {
	  	"timestamp": <...>,
		"machineIdentifier": <...>,
		"processIdentifier": <...>,
		"counter": <...>,
	  },
	  "name": "Product 3",
	  "price": 300
	},
  ]
}
```


With the `extractReferences=false`, `compatibleValues=false` and `objectIdAsMap=false`:

```
{
  "_id": "person",
  "name": "Andrea",
  "bought": ["product1", "product3"]
}
```

Reply:
ok, thanks, can you add the example to the docs?

Reply:
sure